cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -555,6 +556,94 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Update the log level for a program according to the request body. Currently supported program types are
+   * {@link ProgramType#FLOW}, {@link ProgramType#SERVICE} and {@link ProgramType#WORKER}.
+   * The request body is expected to contain a map of log levels, where key is loggername, value is one of the
+   * valid {@link org.apache.twill.api.logging.LogEntry.Level} or null.
+   */
+  @Path("/apps/{app-name}/{program-type}/{program-name}/logLevels")
+  @PUT
+  public void updateProgramLogLevels(HttpRequest request, HttpResponder responder,
+                                     @PathParam("namespace-id") String namespace,
+                                     @PathParam("app-name") String appName,
+                                     @PathParam("program-type") String type,
+                                     @PathParam("program-name") String programName) throws Exception {
+    updateProgramLogLevels(request, responder, namespace, appName, ApplicationId.DEFAULT_VERSION, type, programName);
+  }
+
+  /**
+   * Update the log level for a program according to the request body.
+   */
+  @Path("/apps/{app-name}/versions/{app-version}/{program-type}/{program-name}/logLevels")
+  @PUT
+  public void updateProgramLogLevels(HttpRequest request, HttpResponder responder,
```

Comment:
got through reading the reset of the code and this endpoint is pretty misleading. It really only changes the log level of a running program. If the program is not running, it is a no-op. If you make this call, stop the program, then start it again, it will use its original log level. Was all of this design reviewed?